packages/graphql-hooks/index.d.ts

```diff
@@ -0,0 +1,118 @@
+// Exports
+
+export class GraphQLClient {
+  constructor(options: ClientOptions)
+
+  cache: Cache
+  headers: Headers
+  ssrMode: boolean
+  fetchOptions: object
+  logErrors: boolean
+
+  private onError(): any
+  private fetch(): Promise<any>
+
+  setHeader(key: string, value: string): GraphQLClient
+  setHeaders(object): GraphQLClient
+  logErrorResult({ result: object, operation: Operation }): void
+  request(operation: Operation, options: object): Promise<any>
+}
+
+export function useClientRequest(
+  query: string,
+  initialOptions?: UseClientRequestOptions
```

Comment:
Q: Does this need to be `initialOptions` - to the user, it would just be `options`?

Reply:
`options` works 👍 

```suggestion
  options?: UseClientRequestOptions
```